manifests/plugin/dynflow.pp

```diff
@@ -49,36 +49,40 @@
   }
 
   if $external_core {
-    foreman_proxy::plugin { 'dynflow_core':
-      notify  => Service['smart_proxy_dynflow_core'],
-    }
+    $service = 'smart_proxy_dynflow_core'
 
     file { '/etc/smart_proxy_dynflow_core/settings.yml':
       ensure  => file,
       content => template('foreman_proxy/plugin/dynflow_core.yml.erb'),
       require => Foreman_proxy::Plugin['dynflow_core'],
-      notify  => Service['smart_proxy_dynflow_core'],
+      notify  => Service[$service],
     }
 
     file { '/etc/smart_proxy_dynflow_core/settings.d':
       ensure  => link,
       target  => "${foreman_proxy::config_dir}/settings.d",
       require => Foreman_proxy::Plugin['dynflow_core'],
-      notify  => Service['smart_proxy_dynflow_core'],
+      notify  => Service[$service],
     }
+  } else {
```

Comment:
In the else case, do we want to ensure the settings files are absent to clean it up?

Reply:
I'm not a frequent visitor to the puppet lands so I'm not sure about best practices.

Once we get to the cleanup stage of the proposal, I'd like to drop the handling of those files completely. If we start cleaning those files now, we should probably keep doing that till the end of time, no?

Reply:
I do think it's a good practice to clean up unless a package also places those files. You should keep it in a for a while, but can drop after a few releases. The alternative is that you end up with some cleanup script but why bother if there's already a solution in place that can do exactly that.

And yes, it can live there for a long time. https://github.com/theforeman/puppet-foreman/blob/f69c9f1b90a67798aef4c89e432ee8c29bf05a9b/manifests/config.pp#L51-L54 is proof of that.